Modix.Services/Starboard/StarboardHandler.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Discord;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Modix.Services.Messages.Discord;
+using Modix.Services.Core;
+using Modix.Data.Models.Core;
+using Modix.Services.Quote;
+
+namespace Modix.Services.Starboard
+{
+    public class StarboardHandler :
+        INotificationHandler<ReactionAdded>,
+        INotificationHandler<ReactionRemoved>
+    {
+        private readonly IStarboardService _service;
+        private readonly IDesignatedChannelService _designatedChannelService;
+        private readonly IQuoteService _quoteService;
+
+        public StarboardHandler(
+            IStarboardService service,
+            IDesignatedChannelService designatedChannelService,
+            IQuoteService quoteService)
+        {
+            _service = service;
+            _designatedChannelService = designatedChannelService;
+            _quoteService = quoteService;
+        }
+
+        public Task Handle(ReactionAdded notification, CancellationToken cancellationToken)
+            => HandleReaction(notification.Message, notification.Reaction);
+
+        public Task Handle(ReactionRemoved notification, CancellationToken cancellationToken)
+            => HandleReaction(notification.Message, notification.Reaction);
+
+        private async Task HandleReaction(Cacheable<IUserMessage, ulong> cachedMessage, IReaction reaction)
+        {
+            var emote = reaction.Emote;
+            if (!_service.IsStarEmote(emote))
+            {
+                return;
+            }
+
+            var message = await cachedMessage.GetOrDownloadAsync();
+            if (!(message.Channel is IGuildChannel channel))
+            {
+                return;
+            }
+
+            if(await _designatedChannelService
```

Comment:
Consider moving your conditionals into separate variables. Clustering it like this makes it unreadable.